src/rgw/rgw_rest_s3.cc

```diff
@@ -148,12 +148,21 @@ int RGWGetObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t bl_ofs,
   map<string, string>::iterator riter;
   bufferlist metadata_bl;
 
-  if (op_ret)
-    goto done;
-
   if (sent_header)
     goto send_data;
 
+  if (custom_http_ret) {
```

Comment:
hi @cbodley , when i test radosgw-admin usage , i get object 
```
from boto3.session import Session
import boto3
access_key = "yly"
secret_key = "yly"
url = 'http://127.0.0.1:7480'
session = Session(access_key, secret_key)
s3_client = session.client('s3', endpoint_url=url )

resp = s3_client.get_object(Bucket="test1", Key="512M")
with open('local-backup.bin','wb') as f:
    f.write(resp['Body'].read())
```
before it complete download ,i press ctrl+c to cancal ,and found that the success op =1 and op = 1,  i found the op_ret = -5 , in the current master branch ,it will go send_header and goto send_data,  which will not call set_req_state_err and resorting to 500, and  in the /var/log/ceph/rgw.log file , the message will be GET /bucket/object 200, this is not correct, so should we  remove this patch ?

Reply:
```
2018-08-27 09:50:57.965 7f3efc217700  0 ERROR: flush_read_list(): d->client_cb->handle_data() returned -5
2018-08-27 09:50:57.965 7f3efc217700 20 get_obj_data::cancel_all_io()
2018-08-27 09:50:57.966 7f3efc217700  2 req 1 0.437s s3:get_obj completing
2018-08-27 09:50:57.966 7f3efc217700  2 req 1 0.437s s3:get_obj op status=-5
2018-08-27 09:50:57.966 7f3efc217700  2 req 1 0.437s http status=200
2018-08-27 09:50:57.966 7f3efc217700  1 ====== req done req=0x7f3efc210870 op status=-5 http_status=200 ======
2018-08-27 09:50:57.966 7f3efc217700  1 civetweb: 0x7f3f34f72000: 127.0.0.1 - - [27/Aug/2018:09:50:57 +0800] "GET /ylybucket/1024M HTTP/1.1" 200 321444138 - Boto3/1.8.1 Python/2.7.5 Linux/3.10.0-327.28.3.el7.x86_64 Botocore/1.11.1

```

Reply:
jewel branch
```
2018-08-27 08:15:22.571433 7faa16f25700  1 ====== starting new request req=0x7faa16f1f710 =====
2018-08-27 08:15:22.890051 7faa16f25700  0 ERROR: flush_read_list(): d->client_c->handle_data() returned -5
2018-08-27 08:15:22.893102 7faa16f25700  0 WARNING: set_req_state_err err_no=5 resorting to 500
2018-08-27 08:15:22.893474 7faa16f25700  0 ERROR: s->cio->send_content_length() returned err=-5
2018-08-27 08:15:22.893482 7faa16f25700  0 ERROR: s->cio->print() returned err=-5
2018-08-27 08:15:22.893485 7faa16f25700  0 ERROR: STREAM_IO(s)->print() returned err=-5
2018-08-27 08:15:22.893502 7faa16f25700  0 ERROR: STREAM_IO(s)->complete_header() returned err=-5
2018-08-27 08:15:22.893670 7faa16f25700  1 ====== req done req=0x7faa16f1f710 op status=-5 http_status=500 ======
2018-08-27 08:15:22.893770 7faa16f25700  1 civetweb: 0x7fae5c000bb0: 127.0.0.1 - - [27/Aug/2018:08:15:22 +0800] "GET /ylybucket/100M HTTP/1.1" 500 0 - Boto3/1.7.6 Python/2.7.5 Linux/3.10.0-327.el7.x86_64 Botocore/1.10.6

```

master branch

```
2018-08-27 08:41:59.843 7f26d34f6700  1 ====== starting new request req=0x7f26d34ef870 =====
2018-08-27 08:41:59.895 7f26d34f6700  0 ERROR: flush_read_list(): d->client_cb->handle_data() returned -5
2018-08-27 08:41:59.895 7f26d34f6700  1 ====== req done req=0x7f26d34ef870 op status=-5 http_status=200 ======
2018-08-27 08:41:59.895 7f26d34f6700  1 civetweb: 0x7f270a75c000: 127.0.0.1 - - [27/Aug/2018:08:41:59 +0800] "GET /ylybucket/100M HTTP/1.1" 200 4866108 - Boto3/1.8.1 Python/2.7.5 Linux/3.10.0-327.28.3.el7.x86_64 Botocore/1.11.1


"user": "yly",
            "categories": [
                {
                    "category": "get_obj",
                    "bytes_sent": 4194304,
                    "bytes_received": 0,
                    "ops": 1,
                    "successful_ops": 1
                }
            ],
```

Reply:
https://github.com/ceph/ceph/pull/10767/commits/a084b821839d95b1a12b8bd86429001b44255026 ,   if agree with it ,i will pull requests t fix this,

Reply:
> the message will be GET /bucket/object 200, this is not correct

hi @joke-lee - in this case, haven't we already responded to the client with 200 OK?